### ISSUE TITLE
Inconsistent variable expansion

### ISSUE THREAD

**Issue by atli-c** [firecow/gitlab-ci-local#1451]
**Minimal .gitlab-ci.yml illustrating the issue**
```yml
---
example:
  image: busybox
  environment:
    name: foobar_dev
    url: "http://${CI_ENVIRONMENT_SLUG}.local:3000/"
    action: start
    deployment_tier: development
  script:
    - printenv | sort | grep -E '^(CI_ENVIRONMENT)'
```
The actual script outputs CI_ENVIRONMENT_URL as expected, but one of the last lines from `gitlab-ci-local` itself is missing value from variable in `url: …`

> parsing and downloads finished in 90 ms.
> json schema validated in 135 ms
> example starting busybox:latest (test)
> example copied to docker volumes in 538 ms
> example $ printenv | sort | grep -E '^(CI_ENVIRONMENT)'
> example > CI_ENVIRONMENT_ACTION=start
> example > CI_ENVIRONMENT_NAME=foobar_dev
> example > CI_ENVIRONMENT_SLUG=foobar-dev
> example > CI_ENVIRONMENT_TIER=development
> example > CI_ENVIRONMENT_URL=http://foobar-dev.local:3000/
> example finished in 986 ms
> 
>  PASS  example
> example environment: { name: foobar_dev, url: http://.local:3000/ }
> pipeline finished in 1.33 s

**Expected behavior**
> …
> example environment: { name: foobar_dev, url: http://foobar-dev.local:3000/ }
> …

**Host information**
macOS Sequoia 15.1.1 - Intel x86_64 CPU
gitlab-ci-local 4.56.0

**Containerd binary**
OrbStack

**Additional context**
none

**Comment by ANGkeith** [comment]
fixed by #1443

**Comment by firecow** [comment]
Will be present in next version
